### ISSUE TITLE
urlquery.decode documentation is misleading

### ISSUE THREAD

**Issue by klarose** [open-policy-agent/opa#1592]
# Expected Behavior
urlquery.decode states that it deserializes a "URL query parameter encoded string".

This makes me expect that `urlquery.decode("a=b")` should give `{"a": "b"}`.

# Actual Behavior

 `urlquery.decode("a=b")`  returns `"a=b"`.

See: https://play.openpolicyagent.org/p/0ggpzoLafz

I get: 

```
# Evaluated package in 41.538 µs.
{
  "result": {
    "output": "a=b"
  }
}
```

# Notes

There is another function, `urlquery.encode_object` which takes an object and produces a URL query string. It seems like `urlquery.decode` is really the inverse of `urlquery.encode` which url encodes the given string. The [comment](https://github.com/open-policy-agent/opa/blob/e7af9f82570fd6d2abc81139bc4b692747c2ae57/ast/builtins.go#L844) in the relevant code seems to more accurately describe the behaviour. Perhaps the documentation should use that instead?

Overall, I suspect the documentation is just wrong. That said, it would be nice if the inverse of `urlquery.encode_object` existed. It would make my life easier when applying policies to http requests with relevant information in the query parameters.

In that case, I would expect something like `urlquery.decode_object(urlquery.encode_object({"a":"b","c":"d"}))` to return `{"a": "b", "c": "d"}`.

TL;DR: I think the documentation needs an update, and it'd be nice if we could also have the ability to deserialize an entire URL query string.

**Comment by tsandall** [comment]
@klarose thanks for filing this. I think we should address by updating the docs to clarify that these functions just encode/decode strings to escape URL query parameters. We can also add the decode_object builtin function to avoid repeating the logic required in Rego.
